src/contracts/CrossChainReceiver.sol

```diff
@@ -159,10 +159,21 @@ contract CrossChainReceiver is OwnableWithGuardian, ICrossChainReceiver {
       Errors.CHAIN_ID_MISMATCH
     );
     bytes32 envelopeId = transaction.getEnvelopeId();
-    // if envelope was confirmed before, just return
-    if (_envelopesState[envelopeId] != EnvelopeState.None) return;
 
     bytes32 transactionId = TransactionUtils.getId(encodedTransaction);
+
+    // if envelope was confirmed before, just return
+    if (_envelopesState[envelopeId] != EnvelopeState.None) {
+      emit TransactionReceivedWhenConfirmed(
```

Comment:
Why do we need another type of event?

Reply:
we don't have enough information at the point where we emit to use the other one. See comment above as if we wanted full info then we should do that.